### PR TITLE
feat: better typing for tags

### PR DESCRIPTION
**Pull request by drdelambre**
this makes it a little easier to search in the metric selector by allowing people to type through the equal sign on tags and runs all of the filtering logic through one place

### CODE REVIEW

File: src/flows/pipes/MetricSelector/TagSelectors.tsx [hunk 1]
```diff
@@ -4,18 +4,17 @@ import React, {FC, MouseEvent, useContext, useCallback} from 'react'
...
 import {PipeContext} from 'src/flows/context/pipe'
-import {SchemaContext} from 'src/flows/context/schemaProvider'
+import {NormalizedTag} from 'src/flows/pipes/MetricSelector/context'
 
 // Utils
 import {event as reportEvent} from 'src/cloud/utils/reporting'
 
 type Props = {
-  tags: any[]
+  tags: NormalizedTag[]
```
Comment (Asalem1) [inline src/flows/pipes/MetricSelector/TagSelectors.tsx:13]: 🥳

File: src/flows/pipes/MetricSelector/TagSelectors.tsx [hunk 2]
```diff
@@ -92,55 +91,37 @@ const TagSelectors: FC<Props> = ({tags}) => {
...
-        )
-      })}
-    </>
-  )
+  const _tags = tags.reduce((acc, tag) => {
+    const byTag = Object.entries(tag).reduce((tacc, tcurr) => {
+      const name = tcurr[0]
+      const values = tcurr[1] as string[]
+
+      const items = values.map(val => (
```
Comment (Asalem1) [inline src/flows/pipes/MetricSelector/TagSelectors.tsx:99]: a double reduce map, they do exist!

File: src/flows/pipes/MetricSelector/context.tsx [hunk 3]
```diff
@@ -127,20 +224,21 @@ export const SchemaProvider: FC<Props> = React.memo(({children}) => {
...
 
   useEffect(() => {
-    if (!data?.bucket || loading !== RemoteDataState.NotStarted) {
+    if (!data?.bucket || loading === RemoteDataState.Loading) {
       return
     }
 
     setLoading(RemoteDataState.Loading)
 
+    console.log('neat', data.bucket.name, lastBucket.id, range, query)
```
Comment (Asalem1) [inline src/flows/pipes/MetricSelector/context.tsx]: meh

**Review by Asalem1 — APPROVED** [review]
LGTM. gifs / tests or it didn't happen

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 37 | :red_circle:
accessibility | 96 | :green_circle:
best-practices | 93 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :question:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1015)

**Pull request merged**